### ISSUE TITLE
Genre sometimes not fetched/stored

### ISSUE THREAD

**Issue by timeu** [Marekkon5/onetagger#173]
For some songs the genre tag is not properly stored or fetched although the matched provider contains it. 

Song on beatport: https://www.beatport.com/track/celestial/16796652 
Onetagger log: 
```
2023-01-09 18:12:41 [INFO] onetagger_autotag: Tagging: ~/Music/Nora Van Elken - Celestial.flac
2023-01-09 18:12:47 [INFO] onetagger_autotag: [beatport] State: Ok, Accuracy: Some(1.0), Path: ~/Music/Nora Van Elken - Celestial.flac
```
I checked if there are any settings to enable more verbose debugging logs so one can see which song exactly is matched but I couldn't find anything. 

Also I noticed that for some reason Musicbrainz doesn't return any genres at all. So I had to move it down in the provider list so that a different provider can match and provide the genre information

**Comment by bascurtiz** [comment]
Regarding Beatport (and Genre):
Just tested it with latest build of One Tagger, all good here.
Input was Artist + Title tag only:
![image](https://user-images.githubusercontent.com/34075088/211380152-821f6027-369a-446f-9fb1-3d8af28de45a.png)

Could you try the latest build from the Actions tab?
(Click Actions tab on this Github > Click last build name (one on top) > scroll all the way down + download the artifact for your OS).
Make sure you are logged into your Github account to gain access.

Regarding MusicBrainz (and this track):
This particular track doesn't seem to exist in their database: https://musicbrainz.org/artist/6b6a9b92-ee74-4a3c-91f2-d127911f8812

Regarding MusicBrainz (and genres):
Just tested it with latest build of One Tagger, all good here, except some tracks don't have a genre defined in their database.
Input was Artist + Title tag only:
![image](https://user-images.githubusercontent.com/34075088/211390205-32eb9ff6-ae58-4139-99af-5c618d64a186.png)


Conclusion:
All works fine with the latest build, so I suggest to try that one.
Let me know how that went for you. Thanks!

**Comment by timeu** [comment]
@Marekkon5 : thanks for the quick reply. 
So it also happens with the latest CI/CD build however I think I found out what the issue might be/is. 
I had changed the `GENRES/STYLES OPTIONS` option from `Default` to `Write styles to Genre tag` because on Discogs the styles are more descriptive than the genre.  However this apparently causes some songs on Beatport be have an empty genre tag (If I switch back to the Default setting) I get a proper genre for those songs. 
I was under the impression that the styles only is used on Discogs and it wouldn't affect the other providers. 
Could this be also the reason why all songs matched on MusicBrainz have an empty genre tag ?

**Comment by bascurtiz** [comment]
Yes, not all Beatport tracks in their database have a Style tag, hence.
Most do have a Genre trag however.

Styles is used for Discogs + Beatport. For Discogs it's literally Styles, for Beatport it's the subgenre, they call it iirc.

MusicBrainz doesn't have anything else than Genre, so nope. I think your tracks simply don't exist or have a Genre stated in their db.

**Comment by timeu** [comment]
Thanks, that make sense. 
I wasn't aware that the `GENRES/STYLES OPTIONS` also applies to Beatport's subgenres.
Yeah I think in case of MusicBrainz probably my songs don't have a genre defined (I checked only one of them tough). 

One last questions: Is there somehow an option to increase log verbosity so one can see the tags that were fetched from the provider or at least the url/id of the song ? 
This would make it easier to debug these issues by the user themselves and avoid opening unnecessary issues like this one 🙈

**Comment by bascurtiz** [comment]
Currently we don't have such extended log option, as these could make the logfile 9999x times bigger.
To debug, I advice you to check the URL that's been written into the metadata, to see if there's something either missing on their side or 1T (or a matter of settings).
Make sure to enable the checkbox for URLs under Input & Tags | Select Tags.

Genres/Styles options I tried to decribe in Help section, when hovering over Style / Subgenre:
![image](https://user-images.githubusercontent.com/34075088/211404028-b08ef6dd-9897-45a8-978b-6792105fe7b6.png)

And is stated at Genres/Styles options:
![image](https://user-images.githubusercontent.com/34075088/211404176-5e887e9d-d99a-4744-8d35-6442a1e25420.png)

If this still didn't make you aware, I'm happy to hear from you how you, would like to see it displayed, so we can change it.

**Comment by timeu** [comment]
The extended log option could be set by default to false and only enabled when the need arises to debug a couple of songs that might have issues with tagging. 
It would be sometimes super helpful to see what tags are actually fetched and stored but storing the URL is at least a good workaround to find out what song was matched. I will give it a try.

Regarding the Genre/Styles:  
The help mentions Genre and Style/Subgenre however the option in the advanced settings only mentions Genres/Styles. 
Maybe it's clear that the advanced settings also apply to subgrenes but it might be clearer if the Option is called: `GENRES/STYLES & SUBGENRES OPTIONS`.

**Comment by bascurtiz** [comment]
Good call... What about `GENRE / STYLE / SUBGENRE OPTIONS` ?

**Comment by bascurtiz** [comment]
Demo how I've changed it by now, see video.
https://user-images.githubusercontent.com/34075088/211428223-f818c4ea-25f0-43ce-9804-55a7dee3e91d.mp4

@Marekkon5 will compile it tomorrow, so you can grab the latest build from there, to see the changes by then.

**Comment by Marekkon5** [comment]
Added the UI changes in the latest commit.
